apps/src/accounts/AccountInformation/index.tsx

```diff
@@ -0,0 +1,410 @@
+import classNames from 'classnames';
+import React, {useMemo, useState} from 'react';
+
+import TextField from '@cdo/apps/componentLibrary/textField/TextField';
+import i18n from '@cdo/locale';
+
+import {hashEmail} from '../../code-studio/hashEmail';
+import Alert, {alertTypes} from '../../componentLibrary/alert/Alert';
+import {Button} from '../../componentLibrary/button';
+import {SimpleDropdown} from '../../componentLibrary/dropdown';
+import Link from '../../componentLibrary/link/Link';
+import {Heading2} from '../../componentLibrary/typography';
+import {getAuthenticityToken} from '../../util/AuthenticityTokenStore';
+import ChangeEmailModal from '../ChangeEmailModal';
+
+import {AccountInformationProps} from './types';
+
+import styles from './style.module.scss';
+
+export const AccountInformation: React.FC<AccountInformationProps> = ({
+  authorizedTeacher,
+  secretPictureAccountOnly,
+  teacherManagedAccount,
+  parentManagedAccount,
+  shouldSeeEditEmailLink,
+  isPasswordRequired,
+  isStudent,
+  migrated,
+  userType,
+  userAge,
+  userUsername,
+  userDisplayName,
+  userProperties,
+  userEmail,
+  hashedEmail,
+  encryptedPasswordPresent,
+  canEditPassword,
+  sponsored,
+  ageDropdownOptions,
+  studentInLockoutFlow,
+  showGenderInput,
+  usStateDropdownOptions,
+  countryCode,
+  isUSA,
+}) => {
+  const [name, setName] = useState(userDisplayName ?? '');
```

Comment:
curious why we want these to default to the empty string and not null if they don't exist?

Reply:
eh, makes it a little easier to do in typescript. the backend doesn't seem to save empty strings to the db, I've only seen `null` in there. not sure what rails magic is doing that tbh...

Reply:
ah got it. As long as the empty strings don't get saved, that sounds good to me.